autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/AutofillSettingsViewModel.kt

```diff
@@ -612,6 +664,10 @@ class AutofillSettingsViewModel @Inject constructor(
         object ShowManualCredentialMode : CredentialModeCommand()
     }
 
+    sealed class ListModeCommand(val id: String = UUID.randomUUID().toString()) {
```

Comment:
The id is passed back to the view model in `commandProcessed(command: Command)` function, so that it knows which command in the list is to be removed. so the idea is that the view model will offer up the commands to the activity/fragment to consume but won't know when the view has done so. When the view has done it, it needs to tell the view model it has so it doesn't stay on the list forever. so calling back to `viewModel.commandProcessed` effectively tells the view model "i have processed a command with this ID" and then the viewModel removes it internally.

There were some recommendations around this approach from Google some time back but I can't find the exact ones now... this is as close as I can get to describing this approach: https://developer.android.com/topic/architecture/ui-layer/events#consuming-trigger-updates

instead of IDs and a generic `commandProcessed` function, they would have named functions on the `viewModel` like `snackBarShown()` or `switchedFromViewModeToEditMode` or such....